core-java-modules/core-java-date-operations-3/src/test/java/com/baeldung/date/AddOneMonthToCurrentDateUnitTest.java

```diff
@@ -0,0 +1,79 @@
+package com.baeldung.date;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.time.LocalDate;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+
+import org.apache.commons.lang3.time.DateUtils;
+import org.junit.jupiter.api.Test;
+
+class AddOneMonthToCurrentDateUnitTest {
+
+    @Test
+    void givenCalendarDate_whenAddingOneMonth_thenDateIsChangedCorrectly() {
+        Calendar calendar = Calendar.getInstance();
+        // Dummy current date
+        calendar.set(2023, Calendar.APRIL, 20);
+
+        // add one month
+        calendar.add(Calendar.MONTH, 1);
+
+        assertEquals(Calendar.MAY, calendar.get(Calendar.MONTH));
+        assertEquals(20, calendar.get(Calendar.DAY_OF_MONTH));
+        assertEquals(2023, calendar.get(Calendar.YEAR));
+    }
+
+    @SuppressWarnings("deprecation")
+    @Test
+    void givenDate_whenAddingOneMonth_thenDateIsChangedCorrectly() {
+        // Dummy current date
+        Date currentDate = new Date(2023, 12, 20);
+        // Expected Date
+        Date expectedDate = new Date(2024, 1, 20);
```

Comment:
Probably more meaningful if we use the Calendar constant such as:

```
        Date currentDate = new Date(2023, Calendar.DECEMBER, 20);
        // Expected Date
        Date expectedDate = new Date(2024, Calendar.JANUARY, 20);
```

Reply:
Done